daemon/mgr/container_utils.go

```diff
@@ -251,19 +252,25 @@ func validateResource(r *types.Resources) ([]string, error) {
 
 	// validates memory cgroup value
 	if cgroupInfo.Memory != nil {
-		if r.Memory != 0 && !cgroupInfo.Memory.MemoryLimit {
+		if r.Memory > 0 && !cgroupInfo.Memory.MemoryLimit {
 			warn := "Current Kernel does not support memory limit, discard --memory"
 			logrus.Warn(warn)
 			warnings = append(warnings, warn)
 			r.Memory = 0
 			r.MemorySwap = 0
 		}
-		if r.MemorySwap != 0 && !cgroupInfo.Memory.MemorySwap {
+		if r.MemorySwap > 0 && !cgroupInfo.Memory.MemorySwap {
 			warn := "Current Kernel does not support memory swap, discard --memory-swap"
 			logrus.Warn(warn)
 			warnings = append(warnings, warn)
 			r.MemorySwap = 0
 		}
+		if r.Memory != 0 && r.Memory < MinMemory {
+			return warnings, fmt.Errorf("Minimal memory should greater than 4M")
+		}
+		if r.Memory == 0 && r.MemorySwap > 0 && !update {
```

Comment:
If it is the `create` action, the `--memory-swap` should also work with `memory`, right?

Reply:
Yes, we check only in create, and memory-swap should work with memory